src/content/docs/mobile-monitoring/new-relic-mobile-flutter/monitor-your-flutter-application.mdx

```diff
@@ -0,0 +1,431 @@
+```
+
+3. Add `INTERNET` and `ACCESS_NETWORK_STATE` permissions in your `AndroidManifest.xml` file:
+
+```dart
+<uses-permission android:name="android.permission.INTERNET" />
+<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+```
+
+And you're done! You should start seeing data in New Relic within 5 minutes. 
+
+## View Flutter data in New Relic
+
+Now that you've got the Flutter agent sending data to New Relic, it's time to view, track, and set up alerts for your Flutter app data. 
+
+
+## Troubleshooting
+
+Having trouble finding HTTP data in the UI?
+
+### Problem
+
+After installing the Flutter agent and waiting at least 5 minutes, **No Http data appears:** on the **HTTP errors** and **HTTP requests** UI pages. 
+
+### Solution
+
+If no Http data appears after you wait at least five minutes, check you are not overriding `HttpOverrides.global` inside your Flutter app.
+
+## What's next? 
+
+Check out these optional API calls to further customize how New Relic collects and reports data: 
+
+* [Track errors and resources)](#errors-and-resources)
+* [Track screen navigation events](#screen-nav-events)
+* [GoRouter: Enable automatic routing instrumentation](#gorouter)
+* [Manually report errors](#report-errors)
+
+### Track errors and resources [#errors-and-resources]
+
+set up error tracking and resource tracking. Because NewRelic Mobile Start calls WidgetsFlutterBinding.ensureInitialized, if you are not using NewRelic Mobile Start, you need to call this method prior to calling NewrelicMobile.instance.startAgent.
+
+```dart
+if (Platform.isAndroid) {
+    appToken = AppConfig.androidToken;
+} else if (Platform.isIOS) {
+    appToken = AppConfig.iOSToken;
+}
+
+Config config = Config(
+    accessToken: appToken,
+    analyticsEventEnabled: true,
+    networkErrorRequestEnabled: true,
+    networkRequestEnabled: true,
+    crashReportingEnabled: true,
+    interactionTracingEnabled: true,
+    httpResponseBodyCaptureEnabled: true,
+    loggingEnabled: true,
+    webViewInstrumentation: true,
+    printStatementAsEventsEnabled: true,
+    httpInstrumentationEnabled:true);
+
+// NewrelicMobile.instance.start(config, () {
+//   runApp(MyApp());
+// });
+
+runZonedGuarded(() async {
+    WidgetsFlutterBinding.ensureInitialized();
+    FlutterError.onError = NewrelicMobile.onError;
+    await NewrelicMobile.instance.startAgent(config);
+    runApp(MyApp());
+}, (Object error, StackTrace stackTrace) {
+    NewrelicMobile.instance.recordError(error, stackTrace);
+});
+```
+
+AppToken is platform specific. You need to generate the separate token for Android and iOS apps.
+
+## Track screen navigation events
+
+In order to track navigation events you have to add the NewRelicNavigationObserver to your MaterialApp, WidgetsApp or CupertinoApp.
+
+You should provide a name to route settings: RouteSettings(name: 'Your Route Name'). The root route name / will be replaced by root "/" for clarity's sake.
```

Comment:
@ndesai-newrelic Is this a mandatatory step? If so, I'll remove the "you should" and replace it with something more instructional.